ewwep.py

```diff
@@ -708,15 +701,11 @@ def canAttack(cmd):
 				was_shot = True
 
 			if was_shot:
+				# Flag the user for PvP
+				user_data.time_expirpvp = ewutils.calculatePvpTimer(user_data.time_expirpvp, (int(time.time()) + ewcfg.time_pvp_attack))
 
-				if was_juvenile == True:
-					# Flag the shooter as WANTED no matter what happens next.
-					user_data.time_expirpvp = time_now + ewcfg.time_pvp_kill + int(user_data.bounty / ewcfg.slimecoin_exchangerate)
-
-					user_data.persist()
-
-					# Add the PvP flag role.
-					resp_cont.add_member_to_update(cmd.message.author)
+				user_data.persist()
+				await ewrolemgr.updateRoles(client = cmd.client, member = cmd.message.author)
```

Comment:
doing an await in the middle of such a complicated function is a bad idea. wait for the end to do all awaits. conveniently EwResponseContainer has support for this. just add all members you want to have updated like this: resp_cont.add_member_to_update(cmd.message.author) and await resp_cont.post() at the end of the function